### PR TITLE
Add node module caching for Cypress and Pact tests on GitHub Actions

### PR DESCRIPTION
**Pull request by ddzz**
## Description
https://github.com/department-of-veterans-affairs/va.gov-team/issues/23560

## Testing done
- Tests pass on GHA.

## Acceptance criteria
- [ ] Tests pass on GHA.

### CODE REVIEW

**Review by CBonade — APPROVED** [review]
LGTM!  However, I'd recommend someone with more GHA experience also look it over :)

**Review by U-DON — APPROVED** [review]
Hadn't realized Pact tests were missing the caching step. Good catch. 👍

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -33,6 +33,16 @@ jobs:
         with:
           node-version: ${{ steps.get-node-version.outputs.NODE_VERSION }}
 
+      - name: Cache dependencies
+        id: cache-dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/yarn
+            ~/.cache/Cypress
```
Comment (U-DON) [inline .github/workflows/cypress.yml:42]: Does adding this force the cache to update every time?

Comment (U-DON) [reply .github/workflows/cypress.yml:42]: From what I can tell, missing this hasn't affected the install time much if at all. Without including this in the branch where I consolidated everything into the CI workflow, the install during the Cypress jobs still only takes about 15-25 seconds.

**Pull request merged**